### ISSUE TITLE
abdie "Copy effective config back to abbs source directory if applicable." error for "sudo ciel new" + "sudo ciel build -i main linux-kernel" on Ubuntu-x86_64

### ISSUE THREAD

**Issue by huntercao** [AOSC-Dev/aosc-os-abbs#9000]
On Ubuntu:

> $ uname -a
> Linux hunter-AOC-M7 6.8.0-49-generic #49~22.04.1-Ubuntu SMP PREEMPT_DYNAMIC Wed Nov  6 17:42:15 UTC 2 x86_64 x86_64 x86_64 GNU/Linux

Run `sudo ciel new` create `main` instance. Select default options and run build `sudo ciel build -i main linux-kernel`. There will be error:

> [ERROR]: Effective kernel config is different from source! Features might have been enabled/disabled without getting tracked by abbs source.
> [ERROR]: Updated effective config is at '/var/cache/acbs/build/acbs.el7dzmki/linux-6.11.10/.config-sorted'.
> [INFO]:  Difference as follows:
> --- /var/cache/acbs/build/acbs.el7dzmki/linux-6.11.10/autobuild/amd64/config	2024-12-18 10:06:13.876364189 +0000
> +++ /var/cache/acbs/build/acbs.el7dzmki/linux-6.11.10/.config-sorted	2024-12-18 10:34:19.003475705 +0000
> @@ -1124,6 +1124,8 @@
>  CONFIG_CC10001_ADC=m
>  CONFIG_CC_CAN_LINK_STATIC=y
>  CONFIG_CC_CAN_LINK=y
> +CONFIG_CC_HAS_ASM_GOTO_OUTPUT=y
> +CONFIG_CC_HAS_ASM_GOTO_TIED_OUTPUT=y
>  CONFIG_CC_HAS_ASM_INLINE=y
>  CONFIG_CC_HAS_AUTO_VAR_INIT_PATTERN=y
>  CONFIG_CC_HAS_AUTO_VAR_INIT_ZERO_BARE=y
> @@ -1132,10 +1134,14 @@
>  CONFIG_CC_HAS_IBT=y
>  CONFIG_CC_HAS_INT128=y
>  CONFIG_CC_HAS_KASAN_GENERIC=y
> +CONFIG_CC_HAS_KASAN_SW_TAGS=y
> +CONFIG_CC_HAS_MIN_FUNCTION_ALIGNMENT=y
> +CONFIG_CC_HAS_NAMED_AS_FIXED_SANITIZERS=y
>  CONFIG_CC_HAS_NAMED_AS=y
>  CONFIG_CC_HAS_NO_PROFILE_FN_ATTR=y
>  CONFIG_CC_HAS_RETURN_THUNK=y
>  CONFIG_CC_HAS_SANCOV_TRACE_PC=y
> +CONFIG_CC_HAS_SANE_FUNCTION_ALIGNMENT=y
>  CONFIG_CC_HAS_SANE_STACKPROTECTOR=y
>  CONFIG_CC_HAS_SLS=y
>  CONFIG_CC_HAS_WORKING_NOSANITIZE_ADDRESS=y
> ...
> In file included from /usr/lib/autobuild4/autobuild:13
>                  from /usr/lib/autobuild4/proc/50-build-exec.sh:31
>                  from /usr/lib/autobuild4/templates/00-self.sh:15
> autobuild/build:31: In function `source':
> autobuild/build:31: error: command exited with 0
>    30 | 	echo "$DIFF_OUT"
> >  31 | 	abdie "Copy effective config back to abbs source directory if applicable."
>    32 | fi
> 
> autobuild encountered an error and couldn't continue.

Attached the build log: [acbs-build_jjkdz4aj.log](https://github.com/user-attachments/files/18181489/acbs-build_jjkdz4aj.log)

**Comment by huntercao** [comment]
Here is the .config file on the device in `main` workspace: [config-after-olddefconfig-amd64.txt](https://github.com/user-attachments/files/18183922/config-after-olddefconfig-amd64.txt)

Compare to the [git version config file](https://github.com/AOSC-Dev/aosc-os-abbs/blob/stable/runtime-kernel/linux-kernel/autobuild/amd64/config). There are some configurations missing in our git version.

**Comment by KexyBiscuit** [comment]
Yes, the config in ABBS tree is for building with Core 11, however Core 12 was released recently, I'm going to update the source code soon.

**Comment by KexyBiscuit** [comment]
New kernels are now in stable, this issue should have been fixed, closing.
